Formula/kyuubi.rb

```diff
@@ -0,0 +1,52 @@
+class Kyuubi < Formula
+  desc "Serverless SQL on data warehouses and lakehouses"
+  homepage "https://kyuubi.apache.org/"
+  url "https://www.apache.org/dyn/closer.lua?path=kyuubi/kyuubi-1.6.1-incubating/apache-kyuubi-1.6.1-incubating-bin.tgz"
+  mirror "https://archive.apache.org/dist/kyuubi/kyuubi-1.6.1-incubating/apache-kyuubi-1.6.1-incubating-bin.tgz"
+  version "1.6.1-incubating"
+  sha256 "a64a6e5d71b39b0e494597282493250df1c74714fc2bba5057946926c0a32ba2"
+  license "Apache-2.0"
+  head "https://github.com/apache/kyuubi.git", branch: "master"
+
+  def create_bin_with_envs(target = "kyuubi", source = "kyuubi")
+    (bin/target).write_env_script libexec/"bin"/source,
+      JAVA_HOME:            Language::Java.overridable_java_home_env("1.8")[:JAVA_HOME],
+      SPARK_HOME:           "${SPARK_HOME:-#{Formula["apache-spark"].opt_libexec}}",
+      KYUUBI_HOME:          libexec.to_s,
+      KYUUBI_LOG_DIR:       "${KYUUBI_LOG_DIR:-#{var}/kyuubi/logs}",
+      KYUUBI_PID_DIR:       "${KYUUBI_PID_DIR:-#{var}/kyuubi/pid}",
+      KYUUBI_WORK_DIR_ROOT: "${KYUUBI_WORK_DIR_ROOT:-#{var}/kyuubi/work}"
+  end
```

Comment:
You can put this in the install block. Just make the ENV a variable

Reply:
Can you give me an example?

When I did something like `ENV["JAVA_HOME"] = Formula["openjdk@8"].opt_prefix` in install block. the scripts of kyuubi in `bin` did not respect it

Reply:
```rb
setting = {
      JAVA_HOME:            Language::Java.overridable_java_home_env("1.8")[:JAVA_HOME],
      SPARK_HOME:           "${SPARK_HOME:-#{Formula["apache-spark"].opt_libexec}}",
      KYUUBI_HOME:          libexec.to_s,
      KYUUBI_LOG_DIR:       "${KYUUBI_LOG_DIR:-#{var}/kyuubi/logs}",
      KYUUBI_PID_DIR:       "${KYUUBI_PID_DIR:-#{var}/kyuubi/pid}",
      KYUUBI_WORK_DIR_ROOT: "${KYUUBI_WORK_DIR_ROOT:-#{var}/kyuubi/work}"
}
(bin/"a").write_env_script libexec/"bin/b", setting
(bin/"c").write_env_script libexec/"bin/d", setting
```

Reply:
thanks for the advice @SMillerDev